python/python/lance/dataset.py

```diff
@@ -2489,6 +2490,47 @@ class Transaction:
     blobs_op: Optional[LanceOperation.BaseOperation] = None
 
 
+class Tag(TypedDict):
+    version: int
+    manifest_size: int
+
+
+class Version(TypedDict):
+    version: int
+    timestamp: int | datetime
+    metadata: Dict[str, str]
+
+
+class UpdateResult(TypedDict):
+    num_rows_updated: int
+
+
+class AlterColumn(TypedDict):
+    path: str
+    name: Optional[str]
+    nullable: Optional[bool]
+    data_type: Optional[pa.DataType]
+
+
+class ExecuteResult(TypedDict):
+    num_inserted_rows: int
+    num_updated_rows: int
+    num_deleted_rows: int
```

Comment:
Some of these (e.g. `UpdateResult` and `ExecuteResult`) we had as a typed dict because we thought it might be likely we would add properties in the future and wanted to avoid this being a breaking change.

I wonder if this still holds if we have types declared.  I guess, if this makes it a breaking change then it was technically always a breaking change so maybe it isn't a big deal :laughing:

Reply:
why will it be a break change? in the runtime, typeddict is just dict.

Reply:
Good point.  I guess I'm not really sure why we used typed dict then 😕.  Either way, I'm happy with the change.